uefi/section.go

```diff
@@ -71,18 +84,32 @@ type SectionExtHeader struct {
 	ExtendedSize uint32 `json:"-"`
 }
 
+// SectionGUIDDefined contains the fields for a EFI_SECTION_GUID_DEFINED
+// section.
+type SectionGUIDDefined struct {
+	GUID       uuid.UUID
+	DataOffset uint16
+	Attributes uint16
+}
+
 // Section represents a Firmware File Section
 type Section struct {
 	Header SectionExtHeader
 	Type   string
 	buf    []byte
 
-	//Metadata for extraction and recovery
+	// Metadata for extraction and recovery
 	ExtractPath string
 	fileOrder   int
 
 	// Type specific fields
+	TypeSpecific interface{} `json:",omitempty"`
+
+	// For EFI_SECTION_USER_INTERFACE
 	Name string `json:",omitempty"`
+
+	// Encapsulated firmware
+	Encapsulated Firmware `json:",omitempty"`
```

Comment:
Actually I'm pretty sure I'm looking at something in one of my roms, where there's an lzmacustomdecompress section, with 3 things inside it.

Reply:
In the case of lzma decompression, we should be able to assume that these []Firmware elements are of type Section, but lets keep it as []Firmware so we can reuse the same field for the Volume Image sections as those contain FVs.

Reply:
Generalizing this for []Firmware for all sections also makes the visitor pattern application much nicer I think

Reply:
I didn't know about lzmacustomdecompress. I'll make it an array.

Reply:
Done!

Reply:
We agreed to do this in another PR.